### PR TITLE
Add Mpi tests

### PR DESCRIPTION
**Pull request by TApplencourt**
Only IntelMPI is passing...

### CODE REVIEW

File: xprof/sync_daemon_mpi.c [hunk 1]
```diff
@@ -37,9 +37,11 @@ int MPIX_Init_Session(MPI_Session *lib_shandle, MPI_Comm *lib_comm) {
   /*
    * check we got thread support level foo library needs
    */
+
+  /* Intel MPI doesn't support MPI_Info_get_string
```
Comment (Kerilk) [inline xprof/sync_daemon_mpi.c:41]:
```suggestion
```

File: xprof/sync_daemon_mpi.c [hunk 2]
```diff
@@ -49,6 +51,8 @@ int MPIX_Init_Session(MPI_Session *lib_shandle, MPI_Comm *lib_comm) {
       fprintf(stderr, "THAPI_SYNC_DAEMON_MPI Warning: Did not get MPI_THREAD_SINGLE, got %s\n",
               out_value);
   }
+  */
+
```
Comment (Kerilk) [inline xprof/sync_daemon_mpi.c:55]:
```suggestion
```

File: xprof/sync_daemon_mpi.c [hunk 3]
```diff
@@ -37,9 +37,11 @@ int MPIX_Init_Session(MPI_Session *lib_shandle, MPI_Comm *lib_comm) {
...
    */
+
+  /* Intel MPI doesn't support MPI_Info_get_string
   CHECK_MPI(MPI_Session_get_info(*lib_shandle, &tinfo));
   {
-    char out_value[100]; /* large enough */
+    char out_value[100];
     int valuelen = sizeof(out_value);
     int flag;
     CHECK_MPI(MPI_Info_get_string(tinfo, mt_key, &valuelen, out_value, &flag));
```
Comment (Kerilk) [inline xprof/sync_daemon_mpi.c:47]:
```suggestion
    char out_value[100] = {0};
    int valuelen = sizeof(out_value);
    int flag;
    CHECK_MPI(MPI_Info_get(tinfo, mt_key, &valuelen, out_value, &flag));
```

Comment (Kerilk) [reply xprof/sync_daemon_mpi.c:47]: Wouldn't that work?

File: xprof/sync_daemon_mpi.c [hunk 4]
```diff
@@ -49,6 +51,8 @@ int MPIX_Init_Session(MPI_Session *lib_shandle, MPI_Comm *lib_comm) {
       fprintf(stderr, "THAPI_SYNC_DAEMON_MPI Warning: Did not get MPI_THREAD_SINGLE, got %s\n",
               out_value);
```
Comment (Kerilk) [inline xprof/sync_daemon_mpi.c:52]:
```suggestion
      fprintf(stderr, "THAPI_SYNC_DAEMON_MPI Warning: Did not get %s, got %s\n",
              mt_value, out_value);
```

File: xprof/sync_daemon_mpi.c [hunk 5]
```diff
@@ -49,6 +51,8 @@ int MPIX_Init_Session(MPI_Session *lib_shandle, MPI_Comm *lib_comm) {
       fprintf(stderr, "THAPI_SYNC_DAEMON_MPI Warning: Did not get MPI_THREAD_SINGLE, got %s\n",
```
Comment (Kerilk) [inline xprof/sync_daemon_mpi.c:51]: Should use `else if` above.

Comment (TApplencourt) [reply xprof/sync_daemon_mpi.c]: Fixed with e00c521

File: xprof/sync_daemon_mpi.c (hunk 4 above)
Comment (TApplencourt) [reply xprof/sync_daemon_mpi.c]: Fixed with e00c521

**Comment by TApplencourt** [comment]
```
 # PMPI_Info_get(157): MPI_Info_get(info=0x9c000002, key=thread_level, valuelen=-1596167736, value=0x7ffea0dc6a00, flag=0x7ffea0dc69cc) failed
# PMPI_Info_get(134): Invalid value for valuelen, must be non-negative but is -1596167736
```
I failed something =(

**Comment by TApplencourt** [comment]
Thanks for the fixes!

**Review by Kerilk — APPROVED** [review]

**Pull request merged**